.github/workflows/docker-build-and-publish.yaml

```diff
@@ -62,57 +62,64 @@ on:
+        uses: bakdata/ci-templates/actions/checkout@feat/refactor-docker
         with:
           ref: ${{ inputs.ref }}
-          token: ${{ secrets.github-token }}
-          persist-credentials: false
+          lfs: ${{ inputs.checkout-lfs-files }}
+          submodules: ${{ inputs.checkout-submodules }}
 
-      - name: Build docker image and upload tar file
-        uses: bakdata/ci-templates/actions/docker-build@v1.18.1
-        with:
-          docker-context: "${{ inputs.docker-context }}"
-          dockerfile-path: "${{ inputs.dockerfile-path }}"
-          image-name: "${{ inputs.image-name }}"
-          image-artifact-name: "${{ inputs.image-artifact-name }}"
-          retention-days: "${{ inputs.retention-days }}"
-          working-directory: "${{ inputs.working-directory }}"
-
-
-  docker-publish:
-    name: Docker publish
-    runs-on: ubuntu-22.04
-    needs: [docker-build]
-    steps:
       - name: Login to the Registry
-        uses: "docker/login-action@v2.1.0"
+        uses: "docker/login-action@v3"
         with:
           registry: "${{ inputs.docker-registry }}"
           username: "${{ secrets.docker-user }}"
           password: "${{ secrets.docker-password }}"
 
-      - name: Use uploaded tar files to push new image to docker
-        uses: bakdata/ci-templates/actions/docker-publish@v1.18.1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Set image name
+        run: |
+          fullImageName="${{ inputs.image-name }}"
+          if [[ -n "${{ inputs.image-namespace }}" ]]; then
+            fullImageName="${{ inputs.image-namespace }}/${fullImageName}"
+          fi
+          if [[ -n "${{ inputs.docker-registry }}" ]]; then
+            fullImageName="${{ inputs.docker-registry }}/${fullImageName}"
+          fi
+
+          echo "IMAGE_NAME=${fullImageName}" >> "$GITHUB_ENV"
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ env.IMAGE_NAME }}
+          # on push/PR: dynamically set the CI run id and Git SHA as a custom tag
+          # on tag: release as latest and semver tag
+          tags: |
+            event=push,type=raw,value=${{ inputs.image-tag }}
+            event=pr,type=raw,value=${{ inputs.image-tag }}
+            event=tag,type=semver,pattern={{ version }}
+
+      - name: Build and push
+        uses: docker/build-push-action@v5
         with:
-          docker-registry: "${{ inputs.docker-registry }}"
-          image-namespace: "${{ inputs.image-namespace }}"
-          image-name: "${{ inputs.image-name }}"
-          image-tag: "${{ inputs.image-tag }}"
-          image-artifact-name: "${{ inputs.image-artifact-name }}"
-          working-directory: "${{ inputs.working-directory }}"
+          context: ${{ inputs.docker-context }}
+          file: ${{ inputs.docker-context }}/${{ inputs.dockerfile-path }}
```

Comment:
Is it a valid assumption that the Dockerfile always resides inside the Docker context? I think there can be cases where this is not the case and it would be better not to repeat `inputs.docker-context` here.

Reply:
That's the default from the Docker CLI, not sure why it is defined like this:
```
$ docker build --help
Usage:  docker buildx build [OPTIONS] PATH | URL | -

...

Options:
      ...
  -f, --file string                   Name of the Dockerfile (default: "PATH/Dockerfile")
     ...
```

Should I set the default of the `inputs.dockerfile.path` to `./Dockerfile` instead? This would allow more flexibility, but results in the user having to be aware, that the `file:` is not working like it would locally

Reply:
Sadly defaults of inputs don't allow references to other inputs like this:
```
 dockerfile-path:
        description: "Path to the Dockerfile."
        required: false
        default: "${{ inputs.docker-context }}/Dockerfile"
        type: string
```

Reply:
It's too bad that we can't set `dockerfile-path.default = null` or something for the inputs of this workflow and let the action set the value from its defined default which is `{context}/Dockerfile`... 🤔 

> Should I set the default of the inputs.dockerfile-path to ./Dockerfile instead?
I would prefer this and then setting `file: ${{ inputs.dockerfile-path }}` here as I think it's less confusing. 

(Also, while the Docker CLI uses `PATH/Dockerfile` as _default_, there you can still override it. Here, as it is currently, you cannot set `file` indepdently of the `docker-context` input.)

Reply:
I set the default of `dockerfile-path` to `./Dockerfile` now